src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryFormatter.cs

```diff
@@ -0,0 +1,119 @@
+        {
+            if (serializationStream == null)
+            {
+                throw new ArgumentNullException(nameof(serializationStream), SR.Format(SR.ArgumentNull_WithParamName, serializationStream));
+            }
+            if (serializationStream.CanSeek && (serializationStream.Length == 0))
+            {
+                throw new SerializationException(SR.Serialization_Stream);
+            }
+
+            var formatterEnums = new InternalFE()
+            {
+                _FEtypeFormat = _typeFormat,
+                _FEserializerTypeEnum = InternalSerializerTypeE.Binary,
+                _FEassemblyFormat = _assemblyFormat,
+                _FEsecurityLevel = _securityLevel,
+            };
+
+            var reader = new ObjectReader(serializationStream, _surrogates, _context, formatterEnums, _binder)
+            {
+                _crossAppDomainArray = _crossAppDomainArray
+            };
+            var parser = new BinaryParser(serializationStream, reader);
+            return reader.Deserialize(handler, parser, check);
+        }
+
+        public object Deserialize(Stream serializationStream, HeaderHandler handler) => 
+            Deserialize(serializationStream, handler, check: true);
+
+        [ComVisible(false)]
+        public object UnsafeDeserialize(Stream serializationStream, HeaderHandler handler) => 
+            Deserialize(serializationStream, handler, check: false);
+
+        public void Serialize(Stream serializationStream, object graph) => 
+            Serialize(serializationStream, graph, headers: null);
+
+        public void Serialize(Stream serializationStream, object graph, Header[] headers) => 
+            Serialize(serializationStream, graph, headers, check: true);
+
+        internal void Serialize(Stream serializationStream, object graph, Header[] headers, bool check)
+        {
+            if (serializationStream == null)
+            {
+                throw new ArgumentNullException(nameof(serializationStream), SR.Format(SR.ArgumentNull_WithParamName, serializationStream));
+            }
+
+            var formatterEnums = new InternalFE()
+            {
+                _FEtypeFormat = _typeFormat,
+                _FEserializerTypeEnum = InternalSerializerTypeE.Binary,
+                _FEassemblyFormat = _assemblyFormat,
+            };
+
+            var sow = new ObjectWriter(_surrogates, _context, formatterEnums, _binder);
+            BinaryFormatterWriter binaryWriter = new BinaryFormatterWriter(serializationStream, sow, _typeFormat);
+            sow.Serialize(graph, headers, binaryWriter, check);
+            _crossAppDomainArray = sow._crossAppDomainArray;
+        }
+
+        internal static TypeInformation GetTypeInformation(Type type)
+        {
+            lock (s_typeNameCache)
+            {
+                TypeInformation typeInformation;
+                if (!s_typeNameCache.TryGetValue(type, out typeInformation))
```

Comment:
I assume that the other conversion to ConcurrentDictionary was to avoid still depending on HashTable.  But I saw this one and was going to mention GetOrAdd again, and noticed it was already a generic Dictionary.

And since I had the thought I feel compelled to share it: do we want to use a ConcurrentDictionary here?